File: weeks/01/2_time_conversion/time_conversion.js
function timeConversion(s) {
    let hour = s.slice(0, 2);
    let minute = s.slice(3, 5);
    let second = s.slice(6, 8);
    let ampm = s.slice(8, 10);

    // edge cases
    if (hour === '12' && ampm === 'AM') {
        hour = '00';
    } else if (hour === '12' && ampm === 'PM') {
        // do nothing
    } else {
        if (ampm === 'PM') {
            hour = String(parseInt(hour, 10) + 12);
        }
    }

    return `${hour}:${minute}:${second}`;
}

File: weeks/01/2_time_conversion/time_conversion.rb
def timeConversion(s)
    hour = s[0...2]
    minute = s[3...5]
    second = s[6...8]
    ampm = s[8...10]
  
    # edge cases
    if hour == '12' && ampm == 'AM'
      hour = '00'
    elsif hour == '12' && ampm == 'PM'
      # do nothing
    else
      if ampm == 'PM'
        hour = (hour.to_i + 12).to_s
      end
    end
  
    return "#{hour}:#{minute}:#{second}"
end